develop/dev-guide-implicit-type-conversion.md

```diff
@@ -5,32 +5,32 @@ summary: Introduces the possible consequences of implicit type conversions in Ti
 
-This page introduces the possible consequences of implicit type conversions in TiDB and ways to avoid them.
+This document introduces the rules and possible consequences of implicit type conversions in TiDB and how to avoid implicit type conversions.
 
 ## Conversion rules
 
-When the data types on both sides of a predicate in SQL do not match, TiDB will implicitly convert the data types on one or both sides to a compatible data type for predicate operations.
+When the data types on the two sides of the  predicate in a SQL statement do not match, TiDB implicitly convert the data types on one or both sides to a compatible data type for predicate operations.
 
 The rules for implicit type conversion in TiDB are as follows:
 
-- If one or both arguments are `NULL`, the result of the comparison is `NULL` (except for the NULL-safe `<=>` equivalent comparison operator, which results in `true` for NULL `<=>` NULL and does not require conversion).
+- If one or both arguments are `NULL`, the result of the comparison is `NULL`. The NULL-safe `<=>` equivalent comparison operator does not require conversion, because NULL `<=>` NULL results in `true`.
 - If both arguments in the comparison operation are strings, they are compared as strings.
 - If both arguments are integers, they are compared as integers.
 - If no comparison is made with numbers, the hexadecimal value is treated as a binary string.
-- If one of the arguments is a decimal value, the comparison depends on the other argument. If the other parameter is a decimal or integer value, the parameter is compared to the decimal value, and if the other parameter is a floating-point value, the parameter is compared to the floating-point value.
-- If one of the parameters is a `TIMESTAMP` or `DATETIME` column and the other parameter is a constant, the constant is converted to a timestamp before performing the comparison.
-- In all other cases, the parameters are compared as floating-point numbers (`DOUBLE` type).
+- If one of the arguments is a decimal value, the comparison depends on the other argument. If the other argument is a decimal or integer value, the argument is compared with the decimal value. If the other argument is a floating-point value, the argument is compared with the floating-point value.
+- If one of the arguments is a `TIMESTAMP` or `DATETIME` column and the other argument is a constant, the constant is converted to a timestamp before the comparison is performed.
+- In all other cases, the arguments are compared as floating-point numbers (the `DOUBLE` type).
 
 ## Consequences caused by implicit type conversion
 
-Implicit type conversions increase the ease of human-computer interaction, but we should try to avoid them in application code, due to the fact that they can lead to:
+Implicit type conversions increase the ease of human-computer interaction. However, implicit type conversions need to be avoided in application code, because they might lead to the following issues:
```

Comment:
```suggestion
Implicit type conversions increase the usability of human-computer interaction. However, avoid using implicit type conversions in application code, because they might lead to the following issues:
```